Source/WebCore/platform/xr/openxr/OpenXRInputSource.cpp

```diff
@@ -108,20 +108,13 @@ XrResult OpenXRInputSource::suggestBindings(SuggestedBindings& bindings) const
         RETURN_RESULT_IF_FAILED(createBinding(profile.path, m_pointerAction, makeString(m_subactionPathName, OPENXR_INPUT_AIM_PATH), bindings), m_instance);
 
         // Suggest binding for button actions.
-        WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // WPE port
-        const OpenXRButton* buttons;
-        WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
-        size_t buttonsSize;
-        if (m_handeness == XRHandedness::Left) {
-            buttons = profile.leftButtons;
-            buttonsSize = profile.leftButtonsSize;
-        } else {
-            buttons = profile.rightButtons;
-            buttonsSize = profile.rightButtonsSize;
-        }
+        std::span<const OpenXRButton> buttons;
+        if (m_handeness == XRHandedness::Left)
+            buttons = unsafeMakeSpan(profile.leftButtons, profile.leftButtonsSize);
+        else
+            buttons = unsafeMakeSpan(profile.rightButtons, profile.rightButtonsSize);
```

Comment:
I think that a better way would be to make the members of `struct OpenXRInputProfile` be spans themselves, then there is no need to create temporary spans here. AFAIU it should be possible because spans can be `constexpr` and the values in the profile structs are constants (most of them `constexpr` themselves).